src/compiler/program.ts

```diff
@@ -1321,7 +1321,7 @@ namespace ts {
         }
 
         function getSourceFileByPath(path: Path): SourceFile | undefined {
-            return filesByName.get(path);
+            return options.configFile && toPath(options.configFile.fileName) === path ? options.configFile : filesByName.get(path);

```

Comment:
Can we do this on the project level instead in the program? it just feels wrong to have program return a sourceFile that is not part of it.

Reply:
maybe we should create a new method on Project to getSourceFileOrConfigFile and use that instead.